molecule_hetznercloud/playbooks/create.yml

```diff
@@ -5,153 +5,121 @@
-        server: "{{ item.0.name }}"
-        location: "{{ item.1.location | default(omit) }}"
-        size: "{{ item.1.size | default(10) }}"
-        api_token: "{{ lookup('env', 'HCLOUD_TOKEN') }}"
-        state: "present"
-      loop: "{{ molecule_yml.platforms|subelements('volumes', skip_missing=True)}}"
-      register: volumes
-      async: 7200
-      poll: 0
-
-    - name: Wait for volume(s) creation to complete
-      async_status:
-        jid: "{{ item.ansible_job_id }}"
-      register: hetzner_volumes
-      until: hetzner_volumes.finished
-      retries: 300
-      when:
-        - volumes is defined
-        - volumes.changed
-      with_items: "{{ volumes.results }}"
+      hetzner.hcloud.hcloud_volume:
+        name: "{{ resource_namespace }}-{{ item.name }}"
+        server: "{{ resource_namespace }}-{{ item.server_name }}"
+        size: "{{ item.size | default(10) }}"
+        labels:
+          molecule: "{{ resource_namespace }}"
+        state: present
+      loop: "{{ molecule_yml.platforms | molecule_get_hetznercloud_volumes() }}"
 
-    - name: Create private network(s)
-      hcloud_network:
-        name: "{{ item.name }}"
+    - name: Create network(s)
+      hetzner.hcloud.hcloud_network:
+        name: "{{ resource_namespace }}-{{ item.name }}"
         ip_range: "{{ item.ip_range | default(omit) }}"
-        api_token: "{{ lookup('env', 'HCLOUD_TOKEN') }}"
-        state: "present"
-      loop: "{{ molecule_yml.platforms|molecule_get_hetznercloud_networks('networks') }}"
-      register: networks
-      async: 7200
-      poll: 0
-
-    - name: Wait for network(s) creation to complete
-      async_status:
-        jid: "{{ item.ansible_job_id }}"
-      register: hetzner_networks
-      until: hetzner_networks.finished
-      retries: 300
-      when:
-        - networks is defined
-        - networks.changed
-      with_items: "{{ networks.results }}"
+        labels:
+          molecule: "{{ resource_namespace }}"
+        state: present
+      loop: "{{ molecule_yml.platforms | molecule_get_hetznercloud_networks() }}"
 
-    - name: Create private subnetwork(s)
-      hcloud_subnetwork:
-        network: "{{ item.network_name }}"
-        ip_range: "{{ item.ip|ipaddr('network/prefix') }}"
+    - name: Create subnetwork(s) # noqa jinja[invalid]
+      hetzner.hcloud.hcloud_subnetwork:
+        network: "{{ resource_namespace }}-{{ item.network_name }}"
+        ip_range: "{{ item.ip | ansible.utils.ipaddr('network/prefix') }}"
         network_zone: "{{ item.network_zone | default('eu-central') }}"
         type: "{{ item.type | default('cloud') }}"
-        api_token: "{{ lookup('env', 'HCLOUD_TOKEN') }}"
-        state: "present"
-      loop: "{{ molecule_yml.platforms|molecule_get_hetznercloud_networks('subnetworks') }}"
-      register: subnetworks
+        state: present
+      loop: "{{ molecule_yml.platforms | molecule_get_hetznercloud_subnetworks() }}"
```

Comment:
This tries to attach the server to all subnets, even if their type is not `cloud` or `server` (ie. `vswitch`).

Attaching the server to a vswitch subnet would fail.

Reply:
I added a filter to not attach servers to vswitch networks